### PR TITLE
Buffered channels

### PR DESCRIPTION
**Pull request by joakimmoller**
Using buffered channels since I noticed that the go routings spends a lot of time waiting for synchronization between channels. The value 1000 is a educated guess since I know one picture is about ~2000 rac-records. Innosat channel seems to be the limiting channel.

### CODE REVIEW

**Review by local-minimum — APPROVED** [review]
LGTM, is the size set based on experimenting / random / recommendation? I guess it will set the upper limit of memory usage that may be needed, any risk this might get too high?

File: internal/extractors/extract_data.go [hunk 1]
```diff
@@ -9,13 +9,15 @@ import (
 // ExtractFunction is the type of the ExtractData function
 type ExtractFunction func(callback common.Callback, streamBatch ...StreamBatch)
 
+const channelBufferSize int = 1000
```
Comment (skymandr) [inline internal/extractors/extract_data.go]: Nit-picking here, but are re there any cache chunk related reasons to think that 1024 might be better?

Comment (joakimmoller) [reply internal/extractors/extract_data.go]: OK, it looks better :)

**Comment by joakimmoller** [comment]
@local-minimum: experimenting and guessing - yes, the application will use more heap - we're useing about 120Mb on my laptop (8cpus)

**Comment by joakimmoller** [comment]
Sorry guys I added one more optimization moved the image conversion into the go routine for encoding png.

**Review by local-minimum — CHANGES_REQUESTED** [review]
please undo these extra changes. They will directly conflict with my PR and the optimization here seems much lower priority to me.

File: internal/exports/disk_writer.go [hunk 2]
```diff
@@ -83,7 +83,6 @@ func DiskCallbackFactory(
 					break
 				}
 				imgFileName := getGrayscaleImageName(output, pkg.Origin.Name, ccdImage.PackData)
```
Comment (local-minimum) [inline internal/exports/disk_writer.go:85]:
```suggestion
				imgFileName := getGrayscaleImageName(output, pkg.Origin.Name, ccdImage.PackData)


```

File: internal/exports/disk_writer.go [hunk 3]
```diff
@@ -83,7 +83,6 @@ func DiskCallbackFactory(
 					break
 				}
 				imgFileName := getGrayscaleImageName(output, pkg.Origin.Name, ccdImage.PackData)
-
 				imgData := getImageData(
```
Comment (local-minimum) [inline internal/exports/disk_writer.go]:
```suggestion

				imgData := getImageData(
```

Comment (local-minimum) [reply internal/exports/disk_writer.go]: to keep file untouched.

**Review by local-minimum — APPROVED** [review]

**Pull request merged**